src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -333,6 +340,48 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& options)
         return decltype(ov::internal::caching_properties)::value_type(get_caching_properties());
     }
 
+    // Below properties depend on the device ID.
+    if (name == ov::internal::compiled_model_format.name()) {
+        auto model_format_map = m_compiled_model_format;
+        OPENVINO_ASSERT(m_device_map.find(m_compiled_model_format_device_id) != m_device_map.end(),
+                        "[GPU] compiled_model_format: Couldn't find device for GPU with id ",
+                        m_compiled_model_format_device_id);
+        // Set specified device info for compiled_model_format
+        model_format_map["DEVICE_ID"] = m_compiled_model_format_device_id;
```

Comment:
why do we need to keep `device_id` ? 
`caching_properties` already contain device architecture, which identifies the device we compile for

Reply:
Is there any scenario that device_id changed with the same device if there is multiple GPUs?

Reply:
We can use `driver_info.vendor_id + driver_info.device_id` to guarantee actual device not change?
That means if `driver_info.vendor_id + driver_info.device_id` is same, the device should have the same hardware configuration (cache size, EU number)?

Reply:
> Is there any scenario that device_id changed with the same device if there are multiple GPUs?

yes, if we have multiple discrete GPUs and one of them is removed from that machine between consequent runs.

> We can use driver_info.vendor_id + driver_info.device_id to guarantee actual device not change?

We don't need to guarantee it in `compile_model_format` property. It's a responsibility of `caching_properties`, which holds `ov::device::architecture`, which does not depend on actual order of the devices.

- `caching_properties` defines a list of properties, which describe device and _hardcoded_ compilation options, which affect final blob and its hash
- `compiled_model_format` holds the properties, which can later be changed in runtime (they are not _hardcoded_). 

For example, we compiled out blob and later load it with:
- Different OV runtime version. For plugins like CPU, GPU it's sensitive that OV runtime should be the same, because there are no guarantees that the same blob can be used with multiple OV versions.
- Different driver version (read as execution runtime) for GPU, NPU devices. Drivers can be incompatible with compiled blobs compiled with other drivers (read _compilers_)

Maybe we need a better name for `compiled_model_format` to denote a list of properties, which can be changed in future runs and lead to compiled blob removal (because system is updated and new blob should be triggered).

Reply:
`compiled_model_format` holds some runtime software dependencies for device model running, such as driver version, etc.  While `caching_properties` should hold hardware information, such as architecture, EU number, right?

But we can see that in gpu plugin,  `caching_properties` has contained `ov::intel_gpu::driver_version`, so seems that driver_version is unnecessary in `compiled_model_format` ?

https://github.com/openvinotoolkit/openvino/blob/17008c93895e2bb135246da2a1a3ef23bd43108b/src/plugins/intel_gpu/src/plugin/plugin.cpp#L516-L526

By the way, if we need a proper name for `compiled_model_format`, I vote `compiled_model_runtime_properties`

Reply:
> But we can see that in gpu plugin, caching_properties has contained ov::intel_gpu::driver_version

I suppose we need to move it from `caching_properties`  to `compiled_model_format`. Otherwise, if driver version will be changed, blob hash is changed as well => blob will always be located on the file system and will not be removed by OpenVINO. 
If we move `driver_version` to `compiled_model_format`, then hash is the same, but OV detects that `compiled_model_format` is changed => compiled blob must be dropped and regenerated.

> compiled_model_format, I vote compiled_model_runtime_properties

I agree